src/EnergyPlus/HVACFourPipeBeam.cc

```diff
@@ -0,0 +1,1302 @@
+// C++ Headers
+#include <cmath>
+
+// ObjexxFCL Headers
+#include <ObjexxFCL/Array.functions.hh>
+#include <ObjexxFCL/Fmath.hh>
+
+// EnergyPlus Headers
+#include <AirTerminalUnit.hh>
+#include <HVACFourPipeBeam.hh>
+#include <BranchNodeConnections.hh>
+#include <DataContaminantBalance.hh>
+#include <DataDefineEquip.hh>
+#include <DataEnvironment.hh>
+#include <DataHVACGlobals.hh>
+#include <DataIPShortCuts.hh>
+#include <DataLoopNode.hh>
+#include <DataPlant.hh>
+#include <DataPrecisionGlobals.hh>
+#include <DataSizing.hh>
+#include <DataZoneEnergyDemands.hh>
+#include <DataZoneEquipment.hh>
+#include <FluidProperties.hh>
+#include <General.hh>
+#include <GeneralRoutines.hh>
+#include <InputProcessor.hh>
+#include <NodeInputManager.hh>
+#include <OutputProcessor.hh>
+#include <PlantUtilities.hh>
+#include <Psychrometrics.hh>
+#include <ReportSizingManager.hh>
+#include <ScheduleManager.hh>
+#include <UtilityRoutines.hh>
+#include <CurveManager.hh>
+
+
+namespace EnergyPlus {
+
+namespace FourPipeBeam {
+
+	Array1D< std::shared_ptr< HVACFourPipeBeam > > FourPipeBeams; // dimension to number of machines
```

Comment:
@EnergyArchmage *et al*, we should set up some standards for pointer use before the anti-pattern of using `shared_ptr` everywhere becomes widespread. There is considerable overhead to `shared_ptr` such that it should only be used in contexts where shared ownership is natural.